### ISSUE TITLE
Sokoban inaccessible to people with colorblindness

### ISSUE THREAD

**Issue by wjt** [endlessm/threadbare#1203]
### Description

Here is a typical 3-thread Sokoban puzzle:

<img width="627" height="560" alt="Image" src="https://github.com/user-attachments/assets/010811a2-da4f-4019-bb5d-a2a0fd129556" />

I have full colour vision so I can easily distinguish the three colours of thread.

I used [this GNOME extension](https://extensions.gnome.org/extension/5589/colorblind-filters/) to simulate 3 types of colourblindness on my desktop. (Other kinds of colourblindness are available.) It's clear that in at least some cases the puzzle would be impossible.

One way this problem could be addressed is to change the pattern on each colour of thread so that you do not need colour vision to distinguish them.

### Protanopia (no red perception)

<img width="627" height="560" alt="Image" src="https://github.com/user-attachments/assets/7dfec173-9eb2-4c1a-b6d5-d0bc58fab478" />

The yellow and green threads are not distinguishable.

### Deuteranopia (no green perception)

<img width="627" height="560" alt="Image" src="https://github.com/user-attachments/assets/03dc38e1-f706-45a9-8ca4-75e6ea931a7f" />

The yellow and green threads are not distinguishable.

### Tritanopia (no blue perception)

<img width="627" height="560" alt="Image" src="https://github.com/user-attachments/assets/faf820c0-095a-45ee-acab-d2cdef4ad863" />

All three threads can be distinguished.

### XP Summary (total 300)
| Subdomain | Skill | XP |
| - | - | - |
| Art: Aesthetics & Visual Design | Problem Solving | 150 |
| Art: Aesthetics & Visual Design | Creativity & Design Thinking | 75 |
| Art: Aesthetics & Visual Design | Technical Literacy | 75 |

**Comment by wjt** [comment]
(I tagged & scored this task on the basis of it being addressed by adjusting the design of the threads and spools.)

**Comment by Nilu501** [comment]
I plan to work on this

**Comment by damianslore14** [comment]
Planeo trabajar en esto

**Comment by JuaniCargnel** [comment]
Hi! Damian isn't participating in the classes, and a classmate would like to work on the issue. Could he take it? He will send a message shortly.

**Comment by bergiebird** [comment]
Hello! I'm red/green colorblind, program in Godot, and just found this project. Before I commit to solving this, mind providing some detail on how I may bee-line to this puzzle in-game?

**Comment by wjt** [comment]
Great question! Right now these puzzles are actually not accessible in normal gameplay. (It used to be part of the end-of-quest sequence but we changed that. We had a plan to add them back somewhere else but it's still pending: #2176.)

So the answer is: open `scenes/eternal_loom_sokoban/levels/sokoban_level_1a.tscn` (or one of the other 8 scenes in that folder, `sokoban_level_{1,2,3}{a,b,c}.tscn`) in the editor and hit <kbd>F6</kbd> :)

**Comment by bergiebird** [comment]
<img width="1543" height="1525" alt="Image" src="https://github.com/user-attachments/assets/315b8548-c747-45c4-ba81-6482de5f45bd" />

Thank you wjt! I created a branch titled sokoban_colorblind_fix. (I will try to push it soon)

There are several things everyone should know about colorblindness:
- Colorblindness is uniquely different for each person, but they can still be catagorized into a small handful groups. (Mine is the most common)
- Hue is never affected. Everyone can see more light vs less light.
- The best changes to games for colorblindness involves preserving the artist intent and focusing on only correcting gameplay elements.

My fixes:
- Color Palette untouched
- Yellow is brightest: stripped away a lot of its definition & pattern to make it appear more blinding
- Red is darkest: added harsher definition to accentuate its pattern
- Green is moderate: untouched

My hope:
- I didn't deviate too hard from the artist's intent / game's art style

**Comment by manuq** [comment]
@bergiebird impressive work! And coming from an authorized person. Please open a pull request!

**Comment by manuq** [comment]
@bergiebird, out of curiosity, how did you find this project?

**Comment by bergiebird** [comment]
@manuq I found it through one of StayAtHomeDev's most recent videos! Started perusing the issues mid-watch.

This is my first contribution to any open source project, so I hope I did my PR correctly.

**Comment by manuq** [comment]
@bergiebird thanks for letting us know. And I see that you opened a pull request even! https://github.com/endlessm/threadbare/pull/2943
